DRAFT.md

```diff
@@ -1832,125 +1834,91 @@ the standard if required.
-type to allow constraints to be applied to member functions. This would allow the class
-to be instantiated with `T` as an incomplete type, as member function constraints
-would be checked at only member function instantiation time.
+static_assert(std::is_copy_constructible_v<indirect<Copyable>>); // Passes.
+static_assert(std::is_copy_constructible_v<indirect<NonCopyable>>); // Passes.
+static_assert(std::is_copy_constructible_v<indirect<Incomplete>>); // Passes.
+
+static_assert(std::is_copy_constructible_v<polymorphic<Copyable>>); // Passes.
+static_assert(std::is_copy_constructible_v<polymorphic<NonCopyable>>); // Passes.
+static_assert(std::is_copy_constructible_v<polymorphic<Incomplete>>); // Passes.
+```
 
-For instance:
+### Deferred constraints for incomplete types
+
+It is possible to avoid misleading type trait information by using _deferred
+constraints_: constraint evaluation can be deferred to function instantiation
+time by applying constraints to a deduced template argument.
 
 ```c++
 template <typename T>
-class constrained_wrapper {
+class incomplete_wrapper {
   T* t; // Use a pointer for incomplete type support.
 
  public:
   template <typename TT=T>
-  constrained_wrapper()
+  incomplete_wrapper()
   requires std::is_default_constructible_v<TT>;
 
-  // Constraining the copy constructor is more involved
-  // but possible and omitted for brevity.
-};
-```
-
-By making the default constructor into a function template, and placing the
-constraint on the inferred template type `TT`, evaluation of the constraint
-is deferred to function instantiation time, at which point the type `TT` will be
-deduced to be `T` and will need to be complete for the constraint to be evaluated.
-
-Support for indirect with constraints on a deferred deduced type differs across toolchains
-and inhibits usability. When indirect's member functions are constrained with a deferred deduced
-type, MSVC requires member functions to be explicitly declared so that they can later be defaulted
-when the type is complete. Such behaviour makes using a constrained `indirect` hard to teach and
-inconsistent with the design of current components (like vector) in the standard library.
-
-```c++
-struct Number {}; // Complete type.
+  incomplete_wrapper(const incomplete_wrapper&) requires false;
 
-struct BinOp; // Incomplete type.
-
-struct Expression {
-  std::variant<Number, constrained_indirect<BinOp>> info;
-
-  Expression();
-  Expression(const Expression&);
-  Expression(Expression&&);
-  Expression& operator=(const Expression&);
-  Expression& operator=(Expression&&);
+  template <typename TT=T>
+  incomplete_wrapper(const incomplete_wrapper&)
+  requires std::is_copy_constructible_v<TT>;
 };
 ```
 
-Clang and GCC permit the far more succinct code below.
-
-```c++
-struct Number {}; // Complete type.
+Whilst appealing, this approach works badly in composition with other types.
```

Comment:
I find 'works badly' uncompelling without the context, perhaps 'encounters problems'.

Reply:
How about:

Whilst appealing, this approach is problematic when used in composition with other types.